linera-core/src/remote_node.rs

```diff
@@ -273,6 +273,17 @@ impl<N: ValidatorNode> RemoteNode<N> {
         Ok(hashes)
     }
 
+    #[instrument(level = "trace")]
+    pub async fn download_certificates(
+        &self,
+        hashes: Vec<CryptoHash>,
+    ) -> Result<Vec<Certificate>, NodeError> {
+        if hashes.is_empty() {
+            return Ok(Vec::new());
+        }
```

Comment:
Why not put that into the `download_certificates` itself?

Reply:
I was hesitating to put logic into what is otherwise strictly a thin wrapper about the gRPC and `RpcMessage` interfaces.

The `remote_node` module is exactly the place where we have logic that goes beyond that. Anyway, this check is trivial, of course, so I don't have a strong opinion.